xgcm/grid.py

```diff
@@ -838,6 +838,17 @@ def __init__(
         if metrics is not None:
             self._assign_metrics(metrics)
 
+    def _parse_axes_kwargs(self, kwargs):
+        """Parse string kwargs input to dictionary for each axis.
+        Used e.g. for `boundary`"""
+        parsed_kwargs = dict()
+        if isinstance(kwargs, dict):
+            parsed_kwargs = kwargs
```

Comment:
Do we want to do a consistency check here? Raise an error if the axis is not valid? Or does that happen downstream?

Reply:
In the end the downstream functions like `diff`, `interp` etc will just 'see' `boundary='fill'` for a single axis. The errors should be raised there. Currently axis that do not exist (e.g. if passing `{..., 'Z':{...}}` to a dataset that has only x and y axes) are just silently omitted. Do you think we should raise a warning/error?

Reply:
I think this is a more general question and could be handled in a seperate PR?

Reply:
I think this has been addressed in the test. See `test_missingdim` and `test_missingaxis` in `test_metric_ops_multi_axis.py`. Or do you think we need more fine grained control? Could be nice to have a warning that mentions specifically the argument that is not conform. Let me know.

Reply:
I think it's okay for now.